include/cuco/detail/dynamic_map_kernels.cuh

```diff
@@ -183,7 +184,153 @@ __global__ void insert(InputIt first,
+    submap_block_num_successes[i] = 0;
+  __syncthreads();
+
+  while (it < last) {
+    int i;
+    for (i = 0; i < num_submaps; ++i) {
+      if (submap_mutable_views[i].erase(*it, hash, key_equal)) {
+        atomicAdd(&submap_block_num_successes[i], 1);
+        break;
+      }
+    }
+    it += gridDim.x * blockDim.x;
+  }
+  __syncthreads();
+
+  for (int i = 0; i < num_submaps; ++i) {
+    if (threadIdx.x == 0) {
+      submap_num_successes[i]->fetch_add(static_cast<std::size_t>(submap_block_num_successes[i]),
+                                         cuda::std::memory_order_relaxed);
+    }
+  }
+}
+
+/**
+ * @brief Erases the key/value pairs corresponding to all keys in the range `[first, last)`.
+ *
+ * If the key `*(first + i)` exists in the map, its slot is erased and made available for future
+   insertions.
+ * Else, no effect.
+ * @tparam block_size The size of the thread block
+ * @tparam tile_size The number of threads in the Cooperative Groups used to perform erase
+ * @tparam pair_type Type of the pairs contained in the map
+ * @tparam InputIt Device accessible input iterator whose `value_type` is
+ * convertible to the map's `key_type`
+ * @tparam mutableViewT Type of device view allowing modification of hash map storage
+ * @tparam atomicT Type of atomic storage
+ * @tparam Hash Unary callable type
+ * @tparam KeyEqual Binary callable type
+ * @param first Beginning of the sequence of keys
+ * @param last End of the sequence of keys
+ * @param submap_mutable_views Array of `static_map::mutable_device_view` objects used to
+ * perform `erase` operations on each underlying `static_map`
+ * @param num_successes The number of successfully erased key/value pairs
+ * @param submap_num_successes The number of successfully erased key/value pairs
+ * in each submap
+ * @param num_submaps The number of submaps in the map
+ * @param hash The unary function to apply to hash each key
+ * @param key_equal The binary function to compare two keys for equality
+ */
+template <uint32_t block_size,
+          uint32_t tile_size,
+          typename pair_type,
+          typename InputIt,
+          typename mutableViewT,
+          typename atomicT,
+          typename Hash,
+          typename KeyEqual>
+__global__ void erase(InputIt first,
+                      InputIt last,
+                      mutableViewT* submap_mutable_views,
+                      atomicT** submap_num_successes,
+                      const uint32_t num_submaps,
+                      Hash hash,
+                      KeyEqual key_equal)
+{
+  typedef cub::BlockReduce<std::size_t, block_size> BlockReduce;
+  extern __shared__ unsigned long long submap_block_num_successes[];
+
+  auto tile = cg::tiled_partition<tile_size>(cg::this_thread_block());
+  auto tid  = block_size * blockIdx.x + threadIdx.x;
+  auto it   = first + tid / tile_size;
+
+  for (int i = threadIdx.x; i < num_submaps; i += block_size)
+    submap_block_num_successes[i] = 0;
+  __syncthreads();
```

Comment:
```suggestion
  auto block = cg::this_thread_block();
  auto tile  = cg::tiled_partition<tile_size>(block);
  auto tid   = block_size * block.group_index().x  block.thread_rank();
  auto it    = first + tid / tile_size;

  for (int i = block.thread_rank(); i < num_submaps; i += block_size)
    submap_block_num_successes[i] = 0;
  block.sync();
```
Just a suggestion to use the `cooperative_groups` API whenever possible for better readability.

Reply:
Good idea, added.